deploy/04_Setup_Contracts.js

```diff
@@ -0,0 +1,56 @@
+const { networkConfig, autoFundCheck } = require('../helper-hardhat-config')
+const {ethers, getNamedAccounts} = require('hardhat');
+
+module.exports = async ({
+  getNamedAccounts,
+  deployments
+}) => {
+  const { deploy, log, get } = deployments
+  const chainId = await getChainId()
+  let linkTokenAddress
+  let additionalMessage = ""
+  //set log level to ignore non errors
+  ethers.utils.Logger.setLogLevel(ethers.utils.Logger.levels.ERROR)
+  const networkName = networkConfig[chainId]['name']
+
+  if (chainId == 31337) {
+    linkToken = await get('LinkToken')
+    MockOracle = await get('MockOracle')
+    linkTokenAddress = linkToken.address
+    oracle = MockOracle.address
+    additionalMessage = " --linkaddress " + linkTokenAddress
+  } else {
+    linkTokenAddress = networkConfig[chainId]['linkToken']
+    oracle = networkConfig[chainId]['oracle']
+  }
+
+  //Try Auto-fund APIConsumer contract with LINK
+  const APIConsumer = await deployments.get('APIConsumer')
+  let apiConsumer = await ethers.getContractAt('APIConsumer', APIConsumer.address)  
+
+  let autoFund = await autoFundCheck(apiConsumer.address, networkName, linkTokenAddress, additionalMessage)
+  if (autoFund == true) {
```

Comment:
These can be refactored to
```
if (await autoFundCheck(apiConsumer.address, networkName, linkTokenAddress, additionalMessage)) {
```
instead of having to declare `autoFund` and set its value again later.